lib/sbom/cyclonedx.ex

```diff
@@ -470,4 +465,41 @@ defmodule SBoM.CycloneDX do
     # credo:disable-for-next-line Credo.Check.Warning.UnsafeToAtom
     defp bom_struct(module, unquote(schema_version), attrs), do: struct(Module.concat([unquote(prefix), module]), attrs)
   end
+
+  @spec encode_json(map(), boolean()) :: binary()
+  defp encode_json(encodable, pretty)
+  # Jason selected
+  defp encode_json(encodable, pretty) when @json_module == Jason do
+    if pretty do
+      Jason.encode!(encodable, pretty: true)
+    else
+      Jason.encode!(encodable)
+    end
+  end
+
+  defp encode_json(encodable, pretty) do
+    if pretty do
+      formatter = fn
+        nil, _Enc, _State -> <<"null">>
+        other, enc, state -> :json.format_value(other, enc, state)
+      end
+
+      :json.format(encodable, formatter)
```

Comment:
Careful: JSON polyfills the Erlang json module. If JSON is present, that does not imply that json is as well. Also json.format was introduced later.

All of this means that we have to do some more runtime checking and output good errors if any combination is not available.